lib/net/ssh/proxy/command.rb

```diff
@@ -56,8 +56,8 @@ def open(host, port, connection_options = nil)
       }
       begin
         io = IO.popen(command_line, "r+")
-        if result = Net::SSH::Compat.io_select([io], nil, [io], 60)
-          if result.last.any?
+        if result = self.wait_readable(60)
+          if result == false
```

Comment:
According to http://ruby-doc.org/stdlib-2.0.0/libdoc/io/wait/rdoc/IO.html#method-i-wait_readable it returns IO, true, false or nil. All I've managed to test is, if the timeout triggers, it returns nil. If the wait_readable call is successful, it returns the IO object itself. I assumed that it return false if "the command fails", that is, if the descriptor would be placed in the errors fd array. But to be honest, that's just an assumption. Maybe one should read the bit in ruby source code regarding this?